gstd/src/common/primitives.rs

```diff
@@ -0,0 +1,130 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::errors::{ContractError, Result};
+use crate::prelude::convert::TryFrom;
+use crate::prelude::String;
+use codec::{Decode, Encode};
+use primitive_types::H256;
+use scale_info::TypeInfo;
+
+#[derive(
+    Clone, Copy, Debug, Default, Hash, Ord, PartialEq, PartialOrd, Eq, TypeInfo, Decode, Encode,
+)]
+pub struct ActorId([u8; 32]);
+
+impl ActorId {
+    pub const fn new(arr: [u8; 32]) -> Self {
```

Comment:
I'd leave empty `new()` (same as `default()`) and add `From<[u8; 32]>` implementation or `from_array(...)` method (as it is e.g. for `std::string::String`). As we use `new([0u8; 32])` too often.

Reply:
I don't actually want to give an ability to manipulate with inner array
Is it okay for us?
We may give that ability in case of `debug` feature

Reply:
I don't see the problem if somebody can get the inner value and manipulate with it. But up to you.